lightning/src/ln/outbound_payment.rs

```diff
@@ -951,20 +951,26 @@ impl OutboundPayments {
 			payment_hash, recipient_onion.clone(), keysend_preimage, &route, Some(retry_strategy),
 			payment_params, entropy_source, best_block_height
 		);
-		match self.pending_outbound_payments.lock().unwrap().entry(payment_id) {
-			hash_map::Entry::Occupied(entry) => match entry.get() {
-				PendingOutboundPayment::InvoiceReceived { .. }
-				| PendingOutboundPayment::StaticInvoiceReceived { .. } => {
-					*entry.into_mut() = retryable_payment;
+		let mut invoice_request_opt = None;
+		let mut outbounds = self.pending_outbound_payments.lock().unwrap();
+		match outbounds.entry(payment_id) {
+			hash_map::Entry::Occupied(entry) => match entry.remove() {
+				PendingOutboundPayment::InvoiceReceived { .. } => {
+					outbounds.insert(payment_id, retryable_payment);
```

Comment:
Hmm, I did this to avoid cloning the invoice request. It seemed necessary to fully move the payment out of the map to do so. Or do you mean avoid `remove`ing in the `::InvoiceReceived` case, and keep it in the `::StaticInvoiceReceived` case?

Reply:
Check out the latest version!